internal/raft/message/convert.go

```diff
@@ -0,0 +1,971 @@
+func ConvertCommandToMessageSelect(cmd command.Command) *Command_Select {
+	msgCmdSelect := &Command_Select{}
+	msgCmdSelect.Filter = ConvertCommandToMessageExpr(cmd.(*command.Select).Filter)
+	msgCmdSelect.Input = ConvertCommandToMessageList(cmd.(*command.Select).Input)
+	return msgCmdSelect
+}
+
+// ConvertCommandToMessageCol converts command.Column to a message.Column
+func ConvertCommandToMessageCol(cmd command.Column) *Column {
+	msgCol := &Column{}
+	msgCol.Table = cmd.Table
+	msgCol.Column = ConvertCommandToMessageExpr(cmd.Column)
+	msgCol.Alias = cmd.Alias
+	return msgCol
+}
+
+// ConvertCommandToMessageCols converts []command.Column to a []message.Column
+func ConvertCommandToMessageCols(cmd []command.Column) []*Column {
+	msgCols := []*Column{}
+	for i := range cmd {
+		msgCols = append(msgCols, ConvertCommandToMessageCol(cmd[i]))
+	}
+	return msgCols
+}
+
+// ConvertCommandToMessageProject converts a Command type to a Command_Project type.
+func ConvertCommandToMessageProject(cmd command.Command) *Command_Project {
+	msgCmdProject := &Command_Project{}
+	msgCmdProject.Cols = ConvertCommandToMessageCols(cmd.(*command.Project).Cols)
+	msgCmdProject.Input = ConvertCommandToMessageList(cmd.(*command.Project).Input)
+	return msgCmdProject
+}
+
+// ConvertCommandToMessageDelete converts a Command type to a Command_Delete type.
+func ConvertCommandToMessageDelete(cmd command.Command) *Command_Delete {
+	msgCmdDelete := &Command_Delete{}
+	msgCmdDelete.Table = ConvertCommandToMessageTable(cmd.(*command.Delete).Table)
+	msgCmdDelete.Filter = ConvertCommandToMessageExpr(cmd.(*command.Delete).Filter)
+	return msgCmdDelete
+}
+
+// ConvertCommandToMessageDrop converts a Command type to a CommandDrop type.
+func ConvertCommandToMessageDrop(cmd command.Command) *CommandDrop {
+	msgCmdDrop := &CommandDrop{}
+	switch cmd := cmd.(type) {
+	case *command.DropTable:
+		msgCmdDrop.Target = 0
+		msgCmdDrop.IfExists = cmd.IfExists
+		msgCmdDrop.Schema = cmd.Schema
+		msgCmdDrop.Name = cmd.Name
+	case *command.DropView:
+		msgCmdDrop.Target = 1
+		msgCmdDrop.IfExists = cmd.IfExists
+		msgCmdDrop.Schema = cmd.Schema
+		msgCmdDrop.Name = cmd.Name
+	case *command.DropIndex:
+		msgCmdDrop.Target = 2
+		msgCmdDrop.IfExists = cmd.IfExists
+		msgCmdDrop.Schema = cmd.Schema
+		msgCmdDrop.Name = cmd.Name
+	case *command.DropTrigger:
+		msgCmdDrop.Target = 3
+		msgCmdDrop.IfExists = cmd.IfExists
+		msgCmdDrop.Schema = cmd.Schema
+		msgCmdDrop.Name = cmd.Name
+	}
+	return msgCmdDrop
+}
+
+// ConvertCommandToMessageUpdateOr converts a command.Update or to a message.UpdateOr
+func ConvertCommandToMessageUpdateOr(cmd command.UpdateOr) UpdateOr {
+	switch cmd {
+	case 0:
+		return UpdateOr_UpdateOrUnknown
+	case 1:
+		return UpdateOr_UpdateOrRollback
+	case 2:
+		return UpdateOr_UpdateOrAbort
+	case 3:
+		return UpdateOr_UpdateOrReplace
+	case 4:
+		return UpdateOr_UpdateOrFail
+	case 5:
+		return UpdateOr_UpdateOrIgnore
+	}
+	return -1
```

Comment:
Undocumented return, also magic value?

Reply:
Im not sure what exactly to return here? How can I wrap error on ints?

Reply:
With an additional return argument